### PR TITLE
Implement configurable OIDC claims extraction

### PR DESCRIPTION
**Pull request by ErmakovDmitriy**
This PR expands the authentication agent with methods to extract an OAuth2 token claims and set
their values to HAProxy session variables.

The token claims are prefixed with "token_claim_" and can be used as in an example below:

```
http-request set-header X-OIDC-Username %[var(sess.auth.token_claim_name)] if acl_app3 authenticated
```

#16

### CODE REVIEW

**Comment by mougams** [comment]
Hi @ErmakovDmitriy - sorry for the delay, we missed this PR. Could you please fix conflicts and push back your patch? Thx

**Comment by mougams** [comment]
> Hi @ErmakovDmitriy - sorry for the delay, we missed this PR. Could you please fix conflicts and push back your patch? Thx

@ErmakovDmitriy  I still see out-of-date branch -> could you please rebase and push? Thx

**Comment by ErmakovDmitriy** [comment]
Hi @mougams,

Thank you for your patience!
Please, take a look at the current state. Sorry again for me being clumsy with git.

File: internal/auth/authenticator_oidc.go [hunk 1]
```diff
@@ -328,14 +341,47 @@ func (oa *OIDCAuthenticator) Authenticate(msg *message.Message) (bool, []action.
...
 
+	authorizationURL, err := oa.builaAuthorizationURL(domain, oauthArgs)
+	if err != nil {
+		return false, nil, err
+	}
+
+	return false, []action.Action{BuildRedirectURLMessage(authorizationURL)}, nil
+}
+
+func (oa *OIDCAuthenticator) builaAuthorizationURL(domain string, oauthArgs OAuthArgs) (string, error) {
```
Comment (mougams) [inline internal/auth/authenticator_oidc.go:384]: typo: rename to `buildAuthorizationURL`

**Review by mougams — CHANGES_REQUESTED** [review]
Minor comment - other than that, looks good 

And thanks for having updated the test

File: internal/auth/authenticator_oidc.go (hunk 1 above)
Comment (ErmakovDmitriy) [reply internal/auth/authenticator_oidc.go]: Thank you for noticing it!

Comment (mougams) [reply internal/auth/authenticator_oidc.go]: The tests don't pass, it just hang. Already retriggered twice, same result.

Retrying another time :crossed_fingers:

Comment (mougams) [reply internal/auth/authenticator_oidc.go]: Test passed - I wonder if this is not linked to the HAProxy version - running in the CI, I see version 2.9.9 - and when running locally, I get v2.9.7.

Anyway, not linked to this patch - I'm merging it.

**Review by mougams — APPROVED** [review]

**Pull request merged**